### PR TITLE
Make LTI grader upload tool public

### PR DESCRIPTION
**Pull request by stvstnfrd**
We've been running this at Stanford and would like to share it with the
OpenEdX community.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @stvstnfrd! I've created [OSPR-490](https://openedx.atlassian.net/browse/OSPR-490) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams. 

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ('this must be merged by XX date', and why that is)
- partner information ('this is a course on edx.org')
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the Github pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

We can't start reviewing your pull request until you've added yourself to the [AUTHORS](https://github.com/edx/edx-tools/blob/master/AUTHORS) file. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst) file for more information.

**Comment by stvstnfrd** [comment]
@mondiaz Please take a look over the README instructions; I mostly copy/pasted from the Google doc.

CC: @jinpa

**Comment by sarina** [comment]
@stvstnfrd I'm unfamiliar with this feature, and I imagine many in the community watching the pull requests are too. When opening PRs can you please provide a brief overview of the change/feature contained in the pull request? If you have any documentation or mailing list discussions to link, that's helpful too. See our [tips and suggestions](edx.readthedocs.org/projects/edx-developer-guide/en/latest/process/cover-letter.html) about this.

**Comment by mondiaz** [comment]
@stvstnfrd I looked over the README instructions and noticed some differences, but I'm guessing these are intentional and due to the changes you're making.  Pointing them out just in case:

* Type `%s/^M/\r/g` (for `^M` press `control` and then `m`)
     * When I clean up files, I have to press 'control' + 'v' for '^' and then 'control' + 'm' for 'M'.  Couldn't tell if just 'control' + 'm" worked for me.  Does that remove '^M' for you and add line breaks successfully?
* Is `<./upload_csv.py>` the equivalent of `<python lti_tool.py>` in our platform instructions?
* Is  `<workon lti>` the equivalent of `<workon lti_tool>` in our platform instructions?
* In our platform instructions, we don't need to enter the entire platform URL, just the triplet that is unique to the course in the URL ("organization/coursenumber/courserun").  Does this need to be  https://YOUR_LMS_PLATFORM_URL_HERE in the public instructions?  

Everything else looks good in here.

**Comment by stvstnfrd** [comment]
> I have to press 'control' + 'v' for '^' and then 'control' + 'm' for 'M'.

You are correct; I have fixed this.

> Is <./upload_csv.py> the equivalent of <python lti_tool.py> in our platform instructions?

Yes, it is; the file is renamed and now directly executable.

> Is <workon lti> the equivalent of <workon lti_tool> in our platform instructions?

Yes, the name of a virtualenv is arbitrary; I just shortened it.

> we don't need to enter the entire platform URL

This is the only real change in functionality. The platform URL used to be hard-coded to `class.stanford.edu`. You now need to specify this. Not only does this mean others can use it, but it also means that we can easily run the command against the suclass instance. But you did catch an omission; I forgot to add `COURSE_ID` to that line of the README, which I've now done.

**Comment by stvstnfrd** [comment]
@mondiaz ^

**Comment by stvstnfrd** [comment]
Thanks @sarina 

I went ahead and added a good bit of detail to the `README`, including, but not limited to:
- project overview
- installation instructions
- usage instructions

Please let me know if there's anything I've overlooked!

File: lti/grades/README.markdown [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# LTI Grading
+
+This tool will upload the grades saved in a CSV to an LTI unit created
+in an OpenEdX course.
```
Comment (sarina) [inline lti/grades/README.markdown]: "Open edX" please. https://open.edx.org/trademark-guidelines

**Comment by sarina** [comment]
OK - this seems perfectly reasonable aside from my one comment.

Reading the `README` from the repo, please do update the wiki in the repo with a bit of detail as that requests.

I'll merge this when you and @mondiaz are happy

**Comment by stvstnfrd** [comment]
@sarina Made requested change.

@mondiaz does this all look good to you now?

**Comment by mondiaz** [comment]
@stv This looks good to me now. Thanks!  +1

**Comment by sarina** [comment]
@stvstnfrd - can you be sure to update the wiki as well

**Pull request merged**